nutanix/data_source_nutanix_cluster.go

```diff
@@ -1034,3 +1062,35 @@ func dataSourceNutanixClusterRead(d *schema.ResourceData, meta interface{}) erro
 
 	return nil
 }
+
+func findClusterByName(conn *v3.Client, name string) (*v3.ClusterIntentResponse, error) {
```

Comment:
Nice catch, isn't there any way to find a cluster by name through the API?, If not we can keep like this and create a ticket to the API Tram to support this.